src/app/loadout/Loadouts.tsx

```diff
@@ -79,15 +80,32 @@ export default function LoadoutsContainer({ account }: { account: DestinyAccount
 
 function Loadouts({ account }: { account: DestinyAccount }) {
   const location = useLocation();
+  const dispatch = useDispatch();
+
   const locationStoreId = (location.state as { storeId: string } | undefined)?.storeId;
-  const stores = useSelector(sortedStoresSelector);
-  const currentStore = getCurrentStore(stores)!;
-  const [selectedStoreId, setSelectedStoreId] = useState(
-    locationStoreId && locationStoreId !== 'vault' ? locationStoreId : currentStore.id,
+  const stores = useSelector(storesSelector);
+  const selectedStore = useSelector(selectedLoadoutStore);
+
+  const setSelectedStoreId = useCallback(
+    (storeId: string) => {
+      dispatch(updateLoadoutStore({ storeId }));
+    },
+    [dispatch],
   );
+
+  // Id the location store matches some store id, isn't the vault, and isn't the currently
+  // selected store id we need to dispatch an update to update the page.
+  if (
+    locationStoreId &&
+    stores.some((store) => store.id === locationStoreId) &&
+    locationStoreId !== 'vault' &&
+    selectedStore.id !== locationStoreId
+  ) {
+    setSelectedStoreId(locationStoreId);
+  }
```

Comment:
You shouldn't dispatch actions from within the render function - this needs to be an effect.

After this change it's also probably better if the page that links to loadouts just directly sets the redux state instead of going through location state.

Reply:
I was looking around trying to figure out what page that was. It couldn't.

Reply:
It's in LoadoutPopup.tsx, line 210

Reply:
Legend, thanks

Reply:
> You shouldn't dispatch actions from within the render function - this needs to be an effect.

I actually had in my head that this was the same as `useState` but you are [right](https://github.com/reduxjs/react-redux/discussions/1943#discussioncomment-3426616). It's a shame since that was a nice change to react allowing setting of state in render functions.

Reply:
I don't think you're supposed to call state setters in render functions either.